src/main/java/org/jenkinci/plugins/mock_slave/MockCloud.java

```diff
@@ -63,12 +66,26 @@
     public final Node.Mode mode;
     public final int numExecutors;
     public final String labelString;
+    /** only null when deserializing old configs */
+    private @Nullable Boolean oneShot;
 
-    @DataBoundConstructor public MockCloud(String name, Node.Mode mode, int numExecutors, String labelString) {
+    @DataBoundConstructor public MockCloud(String name, Node.Mode mode, int numExecutors, String labelString, boolean oneShot) {
         super(name);
         this.mode = mode;
         this.numExecutors = numExecutors;
         this.labelString = labelString;
+        this.oneShot = oneShot;
+    }
+
+    public boolean getOneShot() {
+        return oneShot;
```

Comment:
> Implicit conversion without default.

There is no need for a default, because it will not be `null` when this method is called.

> should it not be `isOneShot`?

JavaBeans allows either, and certainly Jenkins databinding is fine with `get`, which is the only usage of this method anyway.